packages/one-app-hmr/lib/html/Document.jsx

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2020 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import { fromJS, Set as iSet } from 'immutable';
+import transit from 'transit-immutable-js';
+
+import { getPublicPath } from '../webpack/utility';
+
+/* eslint-disable react/prop-types */
+export default function Document({
+  modules = [],
+  moduleMap,
+  rootModuleName,
+  lang = 'en-US',
+} = {}) {
+  /* eslint-enable react/prop-types */
+  // TODO: create initial state dynamically on client-side, allow to be user settable
+  const state = transit.toJSON(
+    fromJS({
+      config: {
+        cdnUrl: getPublicPath(),
+        rootModuleName,
+        reportingUrl: '/error',
+      },
+      intl: {
+        activeLocale: lang,
+      },
+      holocron: {
+        loaded: iSet.of(modules.map(({ name }) => name)),
+      },
+    })
+  );
+
+  return (
+    <html lang={lang}>
+      <head>
+        <title>One App Development</title>
+        {React.Children.toArray(
+          modules.filter(({ css }) => !!css).map(({ css }) => <link rel="stylesheet" href={css} />)
```

Comment:
CSS chunks might pose a problem here, we would have tap into `optimization.splitChunks` and group all CSS. chunks into their respective module CSS file.